### ISSUE TITLE
Multiple tab leader issue & sync timers feature

### ISSUE THREAD

**Issue by rowrowrowrow** [SupremeTechnopriest/react-idle-timer#223]
### Affected Module

- `Version`: 5.0.0-rc.16

### Describe the bug

When using two tabs an inactive tab fires onPrompt even though the user is active on the second tab. I was able to confirm that the tab firing the onPrompt is not the leader using the isLeader method in an interval on both tabs.

### To Reproduce

1. Enable cross tab, set a throttle, timeout, prompTimeout, onIdle, onAction, onPrompt, onActive
2. Open two tabs
3. Only be active on one tab
4. Confirm the inactive tab still fires onPrompt once it's timeout expires

### Expected behavior

I expect that if emitOnAllTabs is false only the leader will fire onPrompt, onIdle, etc.

### System Information (please complete the following information)

 - `OS`: macOS
 - `Device`: Mac
 - `Browser Vendor`: firefox
 - `Browser Version`: latest

### Additional context

None

**Comment by SupremeTechnopriest** [comment]
Thanks for the report!  Will get to it this weekend.

**Comment by aburleigh94** [comment]
Also interested in this behavior for my use case, +1

**Comment by SupremeTechnopriest** [comment]
So I think setting `emitOnAllTabs` will achieve what you are looking for.  This will ensure the events are triggered on your users active tab regardless of which tab is the leader.  Making the leader follow the user between active tabs would be difficult and a bit chatty on the messaging channel.  I am still considering it, just let me know if `emitOnAllTabs` covers your use cases.

**Comment by rowrowrowrow** [comment]
Setting `emitOnAllTabs` to true is able to get them closely in sync such that the inactive tab still fires onIdle at roughly the same time that the active tab would fire onIdle. However, in my testing the discrepancy can be relatively large and unpredictable, up to 5 seconds in some cases, if multiple tabs/windows are open and being used. The user will expect that they are the same or near exact same in time. I think something similar to the approach I described in https://github.com/SupremeTechnopriest/react-idle-timer/issues/224 would help bring reduce that margin of error.

**Comment by SupremeTechnopriest** [comment]
The only issue with #224 i can see is that It's going to be extremely chatty on the messaging channel.  It would be fine for browsers that support broadcast channel, but for browsers that do not, the local storage fallback is going to be blocking. We can do some tests using the `message` `onMessage` features to sync actions.  Ill whip up an example later today.

**Comment by rowrowrowrow** [comment]
With emitOnAllTabs set to true I think just adding some extra logic to the send and receive for the start and reset actions is all that would be needed. I can take a closer look but I don't think you would need to change the amount of messages just the content of the messages.

**Comment by SupremeTechnopriest** [comment]
Alright here is the simple proof of concept.  https://codesandbox.io/s/idle-timer-sync-timeout-zsr5de

It works pretty well. Thinking about how I want to include this in the library. Maybe a `syncTimers: number` property where the value is how often you want to sync the timers.

What do you think?

**Comment by rowrowrowrow** [comment]
This seems to be working a lot better, they are definitely closer in sync. Yep, I think a syncTimers prop that throttles the reset message makes sense as well. It'd probably need to be in the eventHandler just like emitOnAction. Thanks for putting this together! 

I can more or less see the changes needed would it help if I started a PR?

**Comment by rowrowrowrow** [comment]
@SupremeTechnopriest see https://github.com/SupremeTechnopriest/react-idle-timer/pull/237 Hope it's a good start.

**Comment by SupremeTechnopriest** [comment]
@rowrowrowrow @aburleigh94,

This is implemented in version `5.0.0-rc.21`!  Let me know how it works out for you. The relevant docs are [here](https://idletimer.dev/docs/api/props#synctimers) but its pretty straight forward like we discussed. 

```ts
import { useIdleTimer, workerTimers } from 'react-idle-timer'

const idleTimer = useIdleTimer({
  timers: workerTimers,
  crossTab: true,
  emitOnAllTabs: true,
  syncTimers: 500,
  ...otherProps
})
```

The value is the throttle for the sync event. (At most once every 500ms in this example).  It's also implemented in the demo so you can play around with it.  You can use the demo on the home page https://idletimer.dev for easy iFrame testing.  If you want to try out actual tabs you can go to the demo page direct.  https://idletimer.dev/demo

I added some test coverage for this, but please reopen this issue if you have any problems.

Very nice feature request.  I think this is going to be widely used!

**Comment by rowrowrowrow** [comment]
Thanks! I'll be sure to try it out.
